src/dwn.ts

```diff
@@ -106,14 +105,16 @@ export class Dwn {
   public async processMessage(tenant: string, rawMessage: RecordsDeleteMessage): Promise<GenericMessageReply>;
   public async processMessage(tenant: string, rawMessage: RecordsQueryMessage): Promise<RecordsQueryReply>;
   public async processMessage(tenant: string, rawMessage: RecordsReadMessage): Promise<RecordsReadReply>;
-  public async processMessage(tenant: string, rawMessage: RecordsWriteMessage, dataStream?: Readable): Promise<GenericMessageReply>;
-  public async processMessage(tenant: string, rawMessage: unknown, dataStream?: Readable): Promise<UnionMessageReply>;
-  public async processMessage(tenant: string, rawMessage: GenericMessage, dataStream?: Readable): Promise<UnionMessageReply> {
+  public async processMessage(tenant: string, rawMessage: RecordsWriteMessage, options?: MessageOptions): Promise<GenericMessageReply>;
```

Comment:
Can we introduce `RecordsWriteMessageOptions` now with the assumption that there will be another param(s) added to the `MessageOption` in the next PR?

Reply:
Yup will do this now! I was going to do it but i figured usually we wait until there's the need for it.